tesseract_environment/src/environment.cpp

```diff
@@ -373,14 +373,15 @@ tesseract_kinematics::KinematicGroup::UPtr Environment::getKinematicGroup(const
   std::shared_lock<std::shared_mutex> lock(mutex_);
 
   std::unique_lock<std::shared_mutex> cache_lock(kinematic_group_cache_mutex_);
-  auto it = kinematic_group_cache_.find(group_name);
+  std::string group_plus_ik_name = group_name + "_" + ik_solver_name;
```

Comment:
I would update the cache to use a `std::pair<std::string, std::string>` as the key.
``` c++
auto key = std::make_pair(group_name, ik_solver_name);
```

Reply:
By default you can't use a std::pair<> as the key to an unordered map. You need to provide a hashing operation for the pair, so in the end I set it up so what's happening under the hood is functionally the same as what you see here.

Reply:
I see, I would switch the cache form an `std::unordered_map`  to a `std::map` then you will not need the hash function.

Reply:
Do you want me to switch all the caches to that for sake of consistency?

Reply:
Nah, those should remain the same because the key is actually a string. From what I understand unordered_map is better performing when the key is a string. In this case we have switched to std::pair as a key so std::map is ok.